create-cluster.rb

```diff
@@ -27,35 +27,39 @@
 MACHINE_TYPES = {
   SMALL => {
     "master_node_machine_type" => "c4.large",
-    "worker_node_machine_type" => "r5.large",
+    "worker_node_machine_type" => "r5.large"
   },
   MEDIUM => {
     "master_node_machine_type" => "c4.2xlarge",
-    "worker_node_machine_type" => "r5.xlarge",
+    "worker_node_machine_type" => "r5.xlarge"
   },
   PRODUCTION => {
     "master_node_machine_type" => "c4.4xlarge",
-    "worker_node_machine_type" => "r5.xlarge",
-  },
+    "worker_node_machine_type" => "r5.xlarge"
+  }
 }
 
 def main(options)
   cluster_name = options[:cluster_name]
   cluster_size = options[:cluster_size]
   vpc_name = options[:vpc_name]
+  gitcrypt_unlock = options[:gitcrypt_unlock]
+  integration_tests = options[:integration_tests]
+  extra_wait = options[:extra_wait]
 
   vpc_name = cluster_name if vpc_name.nil?
   usage if cluster_name.nil? || cluster_size.nil?
 
   check_prerequisites(cluster_name)
 
-  execute "git-crypt unlock"
+  execute "git-crypt unlock" if gitcrypt_unlock == true
 
   create_vpc(vpc_name)
   create_cluster(cluster_name, cluster_size, vpc_name)
   run_kops(cluster_name)
+  sleep(extra_wait) unless extra_wait.nil?
   install_components(cluster_name)
-  run_integration_tests(cluster_name)
+  run_integration_tests(cluster_name) if integration_tests == true
```

Comment:
You could just say `if gitcrypt_unlock` and `if integration_tests` here - you don't need the ` == true`

`nil` is "falsy" in ruby, so you can also say `sleep(extra_wait) if extra_wait`

Reply:
Ahhh cool, thank you. Now improved. Please check